hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkCopyOnWriteTable.java

```diff
@@ -449,4 +455,9 @@ public Iterator<List<WriteStatus>> handleInsert(
     createHandle.write();
     return Collections.singletonList(createHandle.close()).iterator();
   }
+
+  @Override
+  public boolean supportsFileGroupReader() {
```

Comment:
Can we use the engine type to decide the code branch so that we can eliminate this method.

Reply:
ok, if we are aligned on making FG reader the way forward for all engines (so that check will be eliminated at some point). Else method is more maintainable. lets still not let this block the PR.

Reply:
Moved to `HoodieEngineContext`.